otherlibs/stdune/ansi_color.ml

```diff
@@ -1,127 +1,183 @@
+    | Fg_bright_green
+    | Fg_bright_yellow
+    | Fg_bright_blue
+    | Fg_bright_magenta
+    | Fg_bright_cyan
+    | Fg_bright_white
+    | Bg_default
+    | Bg_black
+    | Bg_red
+    | Bg_green
+    | Bg_yellow
+    | Bg_blue
+    | Bg_magenta
+    | Bg_cyan
+    | Bg_white
+    | Bg_bright_black
+    | Bg_bright_red
+    | Bg_bright_green
+    | Bg_bright_yellow
+    | Bg_bright_blue
+    | Bg_bright_magenta
+    | Bg_bright_cyan
+    | Bg_bright_white
+    | Bold
+    | Dim
+    | Italic
+    | Underline
+
+  let to_ansi_int = function
+    | Fg_default -> 39
+    | Fg_black -> 30
+    | Fg_red -> 31
+    | Fg_green -> 32
+    | Fg_yellow -> 33
+    | Fg_blue -> 34
+    | Fg_magenta -> 35
+    | Fg_cyan -> 36
+    | Fg_white -> 37
+    | Fg_bright_black -> 90
+    | Fg_bright_red -> 91
+    | Fg_bright_green -> 92
+    | Fg_bright_yellow -> 93
+    | Fg_bright_blue -> 94
+    | Fg_bright_magenta -> 95
+    | Fg_bright_cyan -> 96
+    | Fg_bright_white -> 97
+    | Bg_default -> 49
+    | Bg_black -> 40
+    | Bg_red -> 41
+    | Bg_green -> 42
+    | Bg_yellow -> 43
+    | Bg_blue -> 44
+    | Bg_magenta -> 45
+    | Bg_cyan -> 46
+    | Bg_white -> 47
+    | Bg_bright_black -> 100
+    | Bg_bright_red -> 101
+    | Bg_bright_green -> 102
+    | Bg_bright_yellow -> 103
+    | Bg_bright_blue -> 104
+    | Bg_bright_magenta -> 105
+    | Bg_bright_cyan -> 106
+    | Bg_bright_white -> 107
+    | Bold -> 1
+    | Dim -> 2
+    | Italic -> 3
+    | Underline -> 4
+
+  let to_dyn s = Dyn.string (string_of_int (to_ansi_int s))
+
+  module Of_ansi_code = struct
+    type nonrec t =
+      | Clear
+      | Unknown
+      | Style of t
+  end
+
+  let of_ansi_code : int -> Of_ansi_code.t = function
+    | 39 -> Style Fg_default
+    | 30 -> Style Fg_black
+    | 31 -> Style Fg_red
+    | 32 -> Style Fg_green
+    | 33 -> Style Fg_yellow
+    | 34 -> Style Fg_blue
+    | 35 -> Style Fg_magenta
+    | 36 -> Style Fg_cyan
+    | 37 -> Style Fg_white
+    | 90 -> Style Fg_bright_black
+    | 91 -> Style Fg_bright_red
+    | 92 -> Style Fg_bright_green
+    | 93 -> Style Fg_bright_yellow
+    | 94 -> Style Fg_bright_blue
+    | 95 -> Style Fg_bright_magenta
+    | 96 -> Style Fg_bright_cyan
+    | 97 -> Style Fg_bright_white
+    | 49 -> Style Bg_default
+    | 40 -> Style Bg_black
+    | 41 -> Style Bg_red
+    | 42 -> Style Bg_green
+    | 43 -> Style Bg_yellow
+    | 44 -> Style Bg_blue
+    | 45 -> Style Bg_magenta
+    | 46 -> Style Bg_cyan
+    | 47 -> Style Bg_white
+    | 100 -> Style Bg_bright_black
+    | 101 -> Style Bg_bright_red
+    | 102 -> Style Bg_bright_green
+    | 103 -> Style Bg_bright_yellow
+    | 104 -> Style Bg_bright_blue
+    | 105 -> Style Bg_bright_magenta
+    | 106 -> Style Bg_bright_cyan
+    | 107 -> Style Bg_bright_white
+    | 1 -> Style Bold
+    | 2 -> Style Dim
+    | 3 -> Style Italic
+    | 4 -> Style Underline
+    | 0 -> Clear
+    | _ -> Unknown
+
+  let of_ansi_string n = of_ansi_code (Int.of_string_exn n)
```

Comment:
The fact that it has a nasty surprise of raising an exception when the argument isn't an integer means that it's not a good general purpose function and should be inlined.

Reply:
Perhaps I should handle that case too (as `Unknown`)?

Reply:
I don't think so. Because we're only going to pass it actual escape code. That's something make sure in the parsing code.

Reply:
I've inlined it in the parsing.